src/clj/medusa/changesets.clj

```diff
@@ -111,6 +111,6 @@
         response (tagsoup/parse build-dirs-url)
         links (elements-by-tag-name response :a)
         build-dirs-links (filter #(.endsWith (get % 2) build-dirs-suffix) links)
-        target-link (first (filter #(.startsWith (get % 2) date) build-dirs-links))
+        target-link (filter #(.startsWith (get % 2) date) build-dirs-links)
```

Comment:
I'm new to Clojure, too, but it seems to me that this will change the sort of thing being returned: whatever was returned before, now a lazy sequence of them get returned:

```clojure
clojure-hello.core=> (first [1 2 3 4 5])
1
clojure-hello.core=> (filter #(= % 4) [1 2 3 4 5])
(4)
```
It's possible that the next line still deals with it correctly, but it's not at all obvious to me. I'm still trying to get things not to crash so I can examine the data and understand what all the `(get % 2)` is about.

Reply:
Got it going. Indeed, in my reduced case, it crashes at `(get target-link 2)`:

```
Exception in thread "main" java.lang.NullPointerException, compiling:(/private/var/folders/39/cttgwlvj0rs_73t1z11pltrm0000gp/T/form-init5428897020641585854.clj:1:125)
	at clojure.lang.Compiler.load(Compiler.java:7391)
	at clojure.lang.Compiler.loadFile(Compiler.java:7317)
	at clojure.main$load_script.invokeStatic(main.clj:275)
	at clojure.main$init_opt.invokeStatic(main.clj:277)
	at clojure.main$init_opt.invoke(main.clj:277)
	at clojure.main$initialize.invokeStatic(main.clj:308)
	at clojure.main$null_opt.invokeStatic(main.clj:342)
	at clojure.main$null_opt.invoke(main.clj:339)
	at clojure.main$main.invokeStatic(main.clj:421)
	at clojure.main$main.doInvoke(main.clj:384)
	at clojure.lang.RestFn.invoke(RestFn.java:421)
	at clojure.lang.Var.invoke(Var.java:383)
	at clojure.lang.AFn.applyToHelper(AFn.java:156)
	at clojure.lang.Var.applyTo(Var.java:700)
	at clojure.main.main(main.java:37)
Caused by: java.lang.NullPointerException
	at java.util.regex.Matcher.getTextLength(Matcher.java:1283)
	at java.util.regex.Matcher.reset(Matcher.java:309)
	at java.util.regex.Matcher.<init>(Matcher.java:229)
	at java.util.regex.Pattern.matcher(Pattern.java:1093)
	at clojure.core$re_matcher.invokeStatic(core.clj:4674)
	at clojure.core$re_find.invokeStatic(core.clj:4716)
	at clojure.core$re_find.invoke(core.clj:4716)
	at clojure_hello.core$find_date_buildid.invokeStatic(core.clj:42)
	at clojure_hello.core$find_date_buildid.invoke(core.clj:32)
	at clojure_hello.core$_main.invokeStatic(core.clj:48)
	at clojure_hello.core$_main.doInvoke(core.clj:45)
	at clojure.lang.RestFn.invoke(RestFn.java:397)
	at clojure.lang.Var.invoke(Var.java:375)
	at user$eval5.invokeStatic(form-init5428897020641585854.clj:1)
	at user$eval5.invoke(form-init5428897020641585854.clj:1)
	at clojure.lang.Compiler.eval(Compiler.java:6927)
	at clojure.lang.Compiler.eval(Compiler.java:6917)
	at clojure.lang.Compiler.load(Compiler.java:7379)
	... 14 more
```

Reply:
So I'd recommend…

1. Pluralizing `find-date-buildid` into `find-date-buildids`
2. Changing `notify-subscribers`, which is its only caller, to expect multiple results
3. Tweaking notify-subscribers' email template a bit to link to multiple changeset URLs if they exist

Of course, I have a whopping 2 days of experience with this language and this code, so I could be wrong. :-)

Reply:
Hey, sounds great. Remind me to get you some keys to production one of these days.